### ISSUE TITLE
Codegen against PostgreSQL < 11 logs a benign error message

### ISSUE THREAD

**Issue by JanecekPetr** [jOOQ/jOOQ#8970]
### Expected behavior and actual behavior:

While generating jOOQ code via the Maven plugin from a running Postgres (class names anonymized):

```
[INFO] Table records generated  : Total: 2.917s, +830.723ms
[INFO] Generating ENUMs
[INFO] Generating ENUM          : Whatever.java
[INFO] Generating ENUM          : SomeOtherEnum.java
[INFO] Enums generated          : Total: 2.944s, +26.727ms
[INFO] Domains fetched          : 0 (0 included, 0 excluded)
ERROR:  column pg_proc.prokind does not exist at character 8
HINT:  Perhaps you meant to reference the column "pg_proc.probin".
STATEMENT:  select "pg_catalog"."pg_proc"."prokind" from "pg_catalog"."pg_proc" where 1 = 0
[INFO] Routines fetched         : 0 (0 included, 0 excluded)
[INFO] Generating routines and table-valued functions
[INFO] Routines generated       : Total: 3.001s, +57.427ms
[INFO] Generation finished: public: Total: 3.002s, +0.367ms
```

The interesting part, obviously, is

>>>
```
ERROR:  column pg_proc.prokind does not exist at character 8
HINT:  Perhaps you meant to reference the column "pg_proc.probin".
STATEMENT:  select "pg_catalog"."pg_proc"."prokind" from "pg_catalog"."pg_proc" where 1 = 0
```
>>>

which, I believe, comes from the fix for #7785: https://github.com/jOOQ/jOOQ/commit/824e74ba9c83dfa905918308450718fdf6958e47#diff-3fb6af070f3401d6becc9ffe5e940e03R752

This suggests that the error message is completely benign and is only a side-effect of checking whether we're on Postgres 11 via feature detection. Still, it'd be nice to get rid of it, or add a descriptive message about it being benign. It did confuse us for a while there.

### Steps to reproduce the problem:

Honestly not sure, it should happen during all Postgres < 11 code generations. If it's hard to reproduce, I'll create a minimal example repo.

### Versions:

- jOOQ: 3.11.11
- Java: 8u201
- Database: Postgres 9.6.3 (don't ask)
- OS: Windows 10
- JDBC Driver: pgjdbc 42.2.2

**Comment by lukaseder** [comment]
Thanks for your report. This should be muted indeed. We have logic to mute any exceptions caused by these version detection queries: https://github.com/jOOQ/jOOQ/blob/version-3.11.11/jOOQ-meta/src/main/java/org/jooq/meta/AbstractDatabase.java#L224

Will check why this is not working (anymore).

**Comment by lukaseder** [comment]
I cannot reproduce this out of the box (I've tried a bogus query on PostgreSQL 11 / latest JDBC driver). The interesting bit is that the log format is different:

```
[INFO] Domains fetched          : 0 (0 included, 0 excluded)
ERROR:  column pg_proc.prokind does not exist at character 8
HINT:  Perhaps you meant to reference the column "pg_proc.probin".
STATEMENT:  select "pg_catalog"."pg_proc"."prokind" from "pg_catalog"."pg_proc" where 1 = 0
```

It appears that in your case, this is being logged with some other logger than the standard one from Maven, e.g. `java.util.logging` or `System.out`? As a workaround, you might be able to mute your drivers log message somehow...

Rather than fixing the logging problem, however, it might be better to write a query that checks the dictionary views for the existence of this `pg_proc.prokind` column, rather than querying the column directly. We can do this in all dialects quite easily. I've created a new, separate issue for this: #8972 

Closing this one as a duplicate of #8972.
